apps/frontend/src/app/PageCharacter/CharacterDisplay/index.tsx

```diff
@@ -82,7 +82,11 @@ export default function CharacterDisplay() {
         fallback={<Skeleton variant="rectangular" width="100%" height={1000} />}
       >
         {characterKey && (
-          <CharacterDisplayCard characterKey={characterKey} onClose={onClose} />
+          <CharacterDisplayCard
+            key={characterKey}
```

Comment:
why does assigning a `key` prevent rerender? curious.

Reply:
Currently we when we switch characters it first renders `CharacterDisplayCard` with the new `characterKey` and old `character` then it rerenders with the new `characterKey` and `character`. The time in between causes bugs
Now it unmounts the old `CharacterDisplayCard` and mounts a fresh copy which prevents this issue